### PR TITLE
Time conversion

### PR DESCRIPTION
**Pull request by SnakeJ419**
No its not inclusive of timezones outside of the us yet, see tom scott video for why i just dont want to deal with it.
Also doesnt only convert to times of those in the server but that will come in a future PR, suggestions on how to structure that are welcome and wanted

### CODE REVIEW

File: src/main/java/Commands/TimeConversion/Time.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        if(hour == 12|| newHour == 12){
+            newIsAM = !newIsAM;
+        }
+
+        return new Time(newHour, minutes, newIsAM, newTimezone);
+    }
+
+    @Override
+    public String toString() {
+        return hour + ":" + (minutes > 9 ? minutes : "0" + minutes) + (isAM ? "am" : "pm");
```
Comment (1Revenger1) [inline src/main/java/Commands/TimeConversion/Time.java:54]: Probably replace this with String.format.
Think it'd be String.format("%d:%02d:%s", hour, minute, isAM ? "Am" : "pm")

**Review by 1Revenger1 — CHANGES_REQUESTED** [review]
Looks mostly fine. We should probably get a linter of some sort.

File: src/main/java/Commands/TimeConversion/Time.java (hunk 1 above)
Comment (SnakeJ419) [reply src/main/java/Commands/TimeConversion/Time.java:54]: oh yeah thats a better call

File: src/main/java/Commands/TimeConversion/ConvertTime.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    private static final String TWELVE_HOUR_REGEX = "(^| )((1[0-2]|0?[1-9])(:[0-5][0-9])?) *[aApP][mM]";
+    private static final Pattern TWELVE_HOUR_PATTERN = Pattern.compile(TWELVE_HOUR_REGEX);
+
+    public ConvertTime(){
+        super("Automagically detects messages containing a time and converts them to timezones for server users");
+    }
+
+    @Override
+    public boolean check(MessageReceivedEvent event, String prefix) {
+        // Figure it out for yourself, I did it once but forgot by the time I went to comment it
```
[the remark this answers is not in the document]
Comment (SnakeJ419) [reply src/main/java/Commands/TimeConversion/ConvertTime.java:24]: it was in comment to the regex expression, it didnt get moved when the regex expression got moved

File: src/main/java/Commands/ServerUtil/ServerUtilPackage.java [hunk 3]
```diff
@@ -14,7 +14,7 @@ public ServerUtilPackage() {
                         new InviteCommand(),
                         new RepeatCommand()
                 }, new SilentEvent[]{}
-        );
+		);
```
[the remark this answers is not in the document]
Comment (SnakeJ419) [reply src/main/java/Commands/ServerUtil/ServerUtilPackage.java:17]: *where are they coming from*

**Review by 1Revenger1 — APPROVED** [review]

**Pull request merged**